src/main/java/cam72cam/immersiverailroading/thirdparty/CommonAPI.java

```diff
@@ -50,26 +102,122 @@ public FluidStack getFluid() {
         return null;
     }
 
+    public Object[] getProperty(String PropertyName) {
+        return getProperty(PropMap.get(PropertyName));
+    }
+
+    public Object[] getProperty(stockProperty PropertyName) {
+        if (stock != null) {
+            switch(PropertyName) {
+                case ID:
+                    return new Object[] { stock.getDefinitionID() };
+                case NAME:
+                    return new Object[] { stock.getName() };
+                case TAG:
+                    return new Object[] { stock.tag };
+                case WEIGET:
+                    return new Object[] { stock.getWeight() };
+                case SPEED:
+                    if (stock instanceof EntityMoveableRollingStock) {
+                        return new Object[] { (EntityMoveableRollingStock)stock.getCurrentSpeed().metric() };
+                    } 
+                    return null;
+                case DIRECTION:
+                    if (stock instanceof EntityMoveableRollingStock) {
+                        double[] SpeedVector = getSpeedVector();
+                        return new Object[] { EnumFacing.fromAngle(SpeedVector[1]).toString()) };
+                    } else {
+                        EnumFacing dir = EnumFacing.fromAngle(stock.getRotationYaw());
+                        return new Object[] { dir.toString() };
+                    }
+                case SPEED_DIRECTION:
+                    if (stock instanceof EntityMoveableRollingStock) {
```

Comment:
Is SPEED_DIRECTION needed?

Reply:
For compatibility with info calls, if you wish get them either.
I suggest keep it.

Reply:
Ok